os-server/src/main/java/app/openschool/auth/AuthServiceImpl.java

```diff
@@ -71,6 +87,33 @@ public UserLoginDto login(String userEmail) {
     return UserLoginMapper.toUserLoginDto(userRepository.findUserByEmail(userEmail));
   }
 
+  @Override
+  public User verifyAccount(VerificationToken verificationToken) {
+    Optional<VerificationToken> fetchedToken =
+        verificationTokenRepository.findVerificationTokenByToken(verificationToken.getToken());
+
+    if (fetchedToken.isPresent()) {
+      if (!verificationToken.isTokenExpired(fetchedToken.get().getCreatedAt(), expiresAt)) {
```

Comment:
why we can't check the expiration time without getting it from the database?

Reply:
In the isTokenExpired method, we get the difference between the token created date and the verification request date, then compare it with the expiration duration. I guess the database request is necessary to get the token's creation date. Or maybe we should save tokens in cache memory?